source/ed/data.cpp

```diff
@@ -218,6 +218,43 @@ void Data::shift_stop_times() {
     }
 }
 
+bool compare(const std::pair<idx_t, size_t>& p1, const std::pair<idx_t, size_t>& p2){
+    return p1.second < p2.second;
+}
+
+void Data::build_route_destination(){
+    // <Route.idx, std::map<StopArea.idx, Count>>
+    std::unordered_map<idx_t, std::map<idx_t, size_t>> destinations;
+    for(auto vj : vehicle_journeys){
+        if((vj->tmp_route) && (!vj->tmp_route->destination)){
+            if (!vj->stop_time_list.empty()){
+                idx_t idx = nt::invalid_idx;
+                if ((vj->stop_time_list.back()->journey_pattern_point)
+                        && (vj->stop_time_list.back()->journey_pattern_point->stop_point)
+                        && (vj->stop_time_list.back()->journey_pattern_point->stop_point->stop_area)){
+                    idx = vj->stop_time_list.back()->journey_pattern_point->stop_point->stop_area->idx;
+                }
+                if (idx != nt::invalid_idx){
```

Comment:
Instead of using inbricated ifs in for loop, I find more clear to continue to the negated condition:
```c++
for (const auto* vj : vehicle_journeys) {
    if (! vj->tmp_route || vj->stop_time_list.empty()) { continue; }
    if (vj->tmp_route->destination) { continue; } // we have a destination, don't create one
    if (! vj->stop_time_list.back()->journey_pattern_point
        || ! vj->stop_time_list.back()->journey_pattern_point->stop_point
        || ! vj->stop_time_list.back()->journey_pattern_point->stop_point->stop_area) {
        continue;
    }
    const auto sa_idx = vj->stop_time_list.back()->journey_pattern_point->stop_point->stop_area->idx;
    ++destinations[vj->tmp_route->idx][sa_idx];
}
```
If the element does not exist, when using [] on a map, a default constructed element will be added, as you want it. And for a size_t, default is 0, all good :+1: